src/components/chat/chat/EmojiPicker.tsx

```diff
@@ -0,0 +1,52 @@
+import { ActionMap, Container } from 'constate'
+import { Picker } from 'emoji-mart'
+import * as React from 'react'
+import styled from 'react-emotion'
+import { Button } from 'semantic-ui-react'
+import { emojiMartStyles } from './_emojimartstyles'
+
+const Wrapper = styled('div')({
+  position: 'absolute',
+  top: '-200%',
+  left: '100%',
+})
+
+interface EmojiPickerProps {
+  selectEmoji: (emoji: string) => void
+}
+
+interface State {
+  open: boolean
+}
+
+interface Actions {
+  setIsOpen: (open: boolean) => void
+}
+
+const actions: ActionMap<State, Actions> = {
+  setIsOpen: (open: boolean) => (state) => ({ ...state, open }),
```

Comment:
i think `constate` merges the state for us, so technically we could just do `({ open })`

Reply:
I think you're right, this is probably cruft.

```suggestion
  setIsOpen: (open: boolean) => () => ({ open }),
```